src/execution/vm/bytecode_generator.cpp

```diff
@@ -2350,6 +2350,17 @@ void BytecodeGenerator::VisitBuiltinStringCall(ast::CallExpr *call, ast::Builtin
       }
       break;
     }
+    case ast::Builtin::Concat: {
+      const auto num_inputs = call->NumArgs() - 1;
+      LocalVar num_inputs_var =
+          GetCurrentFunction()->NewLocal(ast::BuiltinType::Get(call->GetType()->GetContext(), ast::BuiltinType::Int64));
+      GetEmitter()->EmitAssignImm8(num_inputs_var, static_cast<int64_t>(num_inputs));
```

Comment:
The length is a compile-time constant, directly emit the value in the bytecode rather than assigning to a temporary. Also, using an 8-byte count may be overkill?

Reply:
When you say it's a compile-time constant you're talking about when compiling the tpl, not when compiling the C++ right? How would I emit the number directly in the bytecode?

Reply:
> When you say it's a compile-time constant you're talking about when compiling the tpl, not when compiling the C++ right?

Correct.

> How would I emit the number directly in the bytecode?

Take a look at how `EmitDerefN()` works. You should create a new function `BytecodeEmitter::EmitConcat()` that takes in the argument count as a `uint32_t`.

Reply:
Updated